src/app/shared/services/skin/skin.service.ts

```diff
@@ -81,7 +83,7 @@ export class SkinService extends SyncServiceBase {
    */
   private generateOverrideConfig(skin: IAppSkin) {
     // Merge onto new object to avoid changing stored revertOverride
-    const base: RecursivePartial<IAppConfig> = {};
+    const base: RecursivePartial<IAppConfig> = this.deploymentService.config.app_config || {};
```

Comment:
Without this, any deployment config-level overrides to the default config that are not explicitly specified or overwritten in the skin are lost on `setSkin`. E.g., when testing the `debug_skin_select` template on the `debug` deployment, switching skins causes the footer to be removed.

Reply:
Ah, good catch. Diving more deeply it doesn't appear to be the skin service making changes that lose this configuration but the appConfigService that doesn't initialise with it properly. I'll add a specific test to the skin service to verify the case you've presented and fix the appConfigService accordingly (should probably add a few tests to the appConfigService too, having tests are helping massively to debug the skin service now, thanks for adding the updated tests too)

Reply:
See updates in [f1f6fcb](https://github.com/IDEMSInternational/open-app-builder/pull/2423/commits/f1f6fcbf76e03012451a3f0bf94c9b70d12e2779)

Reply:
I've also added a PR to update the debug config for improved testing:
https://github.com/IDEMSInternational/app-debug-content/pull/84